python/dgl/data/qm9.py

```diff
@@ -0,0 +1,127 @@
+"""QM9 dataset for graph property prediction (regression)."""
+import os
+import numpy as np
+import scipy.sparse as sp
+import torch
+
+from .dgl_dataset import DGLDataset
+from .utils import download
+from ..convert import graph as dgl_graph
+from ..transform import to_bidirected
+
+class QM9Dataset(DGLDataset):
+    r"""QM9 dataset for graph property prediction (regression)
+
+    This dataset consists of 13,0831 molecules with 12 regression targets.
+    - id: [0, 1, ..., 133884], a unique identifier for each molecule.
+    - R: [2358210, 3], the coordinates of each atom.
+    - Z: [2358210, ], the atomic number.
+    - N: [130831, ], the number of atoms in each molecule. Atomic properties like Z and R are just concatenated globally, so you need this value to select the correct atoms for each molecule.
+
+    Statistics:
+
+    - Number of graphs: 13,0831
+    - Number of regression targets: 12
+
+    Parameters
+    ----------
+    raw_dir : str
+        Raw file directory to download/contains the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+        Whether to print out progress information. Default: True.
+
+    Attributes
+    ----------
+    num_labels : int
+        Number of labels for each graph, i.e. number of prediction tasks
+
+    Raises
+    ------
+    UserWarning
+        If the raw data is changed in the remote server by the author.
+    
+    Examples
+    --------
+    >>> data = QM9Dataset()
+    >>> data.num_labels
+    12
+    >>>
+    >>> # iterate over the dataset
+    >>> for g, label in data:
+    ...     # get feature
```

Comment:
You need to include the code for feature fetching if features are already in `g`. Otherwise just remove this LoC.